google-cloud-storage/src/main/java/com/google/cloud/storage/StorageImpl.java

```diff
@@ -1483,96 +1483,48 @@ public Notification createNotification(
       final String bucket, final NotificationInfo notificationInfo) {
     final com.google.api.services.storage.model.Notification notificationPb =
         codecs.notificationInfo().encode(notificationInfo);
-    try {
-      return codecs
-          .notificationInfo()
-          .decode(
-              runWithRetries(
-                  new Callable<com.google.api.services.storage.model.Notification>() {
-                    @Override
-                    public com.google.api.services.storage.model.Notification call() {
-                      return storageRpc.createNotification(bucket, notificationPb);
-                    }
-                  },
-                  getOptions().getRetrySettings(),
-                  EXCEPTION_HANDLER,
-                  getOptions().getClock()))
-          .asNotification(this);
-    } catch (RetryHelperException e) {
-      throw StorageException.translateAndThrow(e);
-    }
+    ResultRetryAlgorithm<?> algorithm =
+        retryAlgorithmManager.getForNotificationCreate(bucket, notificationPb);
+    return run(
+        algorithm,
+        () -> storageRpc.createNotification(bucket, notificationPb),
+        codecs.notificationInfo()::decode
+    ).asNotification(this);
```

Comment:
Not incorrect, but my personal preference is to keep the `.as*(this)` next to the decode. My perspective is that the transformation is more an attribute of the transformation from apiary result to our model than post process of result from `run`.

```suggestion
    return run(
        algorithm,
        () -> storageRpc.createNotification(bucket, notificationPb),
        n -> codecs.notificationInfo().decode(n).asNotification(this)
    );
```